Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp

```diff
@@ -678,6 +678,38 @@ void GStreamerRegistryScanner::initializeEncoders(const GStreamerRegistryScanner
     }
 }
 
+GStreamerRegistryScanner::CodecLookupResult GStreamerRegistryScanner::isCapsSupported(Configuration configuration, const char* capsString, bool shouldCheckForHardwareUse) const
+{
+    OptionSet<ElementFactories::Type> factoryTypes;
+    switch (configuration) {
+    case Configuration::Decoding:
+        factoryTypes.add(ElementFactories::Type::VideoDecoder);
+        break;
+    case Configuration::Encoding:
+        factoryTypes.add(ElementFactories::Type::VideoEncoder);
+        break;
+    }
+    auto lookupResult = ElementFactories(factoryTypes).hasElementForMediaType(factoryTypes.toSingleValue().value(), capsString, ElementFactories::CheckHardwareClassifier::Yes);
+    bool supported = lookupResult && shouldCheckForHardwareUse ? lookupResult.isUsingHardware : true;
```

Comment:
Two comments about this line:

* Less important: variable name should include a verb. You can probably shadow the own method name by using `areCapsSupported`.
* More important: If lookupResult is false you are end up assigning `supported` as true if precedence of grouping begins from the left (which, I'm never sure of). So you better add () around `shouldCheckForHardwareUse ? lookupResult.isUsingHardware : true`.